src/components/Nav/index.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import { Drawer, AppBar, Box, Toolbar, IconButton, Typography, Container, Avatar, Tooltip, Button } from '@mui/material';
+import TextField from '@mui/material/TextField';
+import PersonOutlineIcon from '@mui/icons-material/PersonOutline';
+import logoLightMode from '../../assets/images/fit_trax_app_logo_lightmode.png';
+import { avatarStyle, drawerStyle, iconButtonStyle, navStyle, logoStyle, boxStyle, buttonStyle, drawerPaperStyle } from './styles';
+
+const Nav = () => {
+  const [openDrawer, setOpenDrawer] = React.useState(false);
+  const [username, setUsername] = React.useState('');
+  const [age, setAge] = React.useState('');
+  const [fitness, setFitness] = React.useState('');
+  const [gender, setGender] = React.useState('');
+
+  const toggleDrawer = (open: boolean) => (event: React.KeyboardEvent | React.MouseEvent) => {
+    if (event.type === 'keydown' && ((event as React.KeyboardEvent).key === 'Tab' || (event as React.KeyboardEvent).key === 'Shift')) {
+      return;
+    }
+    setOpenDrawer(open);
+  };
+
+  const handleFormSubmit = () => {
+    /**
+     * @TODO: Waiting for API to submit the form
+     */
+    console.log('form gender :>> ', gender);
+    console.log('form age :>> ', age);
+    console.log('form fitness :>> ', fitness);
+    console.log('form username :>> ', username);
+    setOpenDrawer(false);
+  };
+
+  const handleFormCancel = () => {
+    setOpenDrawer(false);
+  };
+
+  return (
+    <AppBar sx={navStyle} position="static">
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          <Typography variant="h6" noWrap component="div" sx={{ flexGrow: 1 }}>
+            <img src={logoLightMode} alt="fit_trax_app_logo" style={logoStyle} />
+          </Typography>
+
+          <Box sx={{ flexGrow: 0 }}>
+            <Tooltip title="User Account">
+              <IconButton onClick={toggleDrawer(true)} sx={iconButtonStyle} data-testid="avatar">
+                <Avatar sx={avatarStyle}>
+                  <PersonOutlineIcon sx={{ fontSize: 45 }} />
+                </Avatar>
+              </IconButton>
+            </Tooltip>
+          </Box>
+
+          <Drawer
+            PaperProps={{
+              sx: drawerPaperStyle,
+            }}
+            anchor="right"
+            open={openDrawer}
+            onClose={toggleDrawer(false)}
+          >
+            <Box component="form" sx={drawerStyle} noValidate autoComplete="off">
+              <h3>User Account</h3>
+
+              <Box sx={boxStyle}>
```

Comment:
you can try mapping an array for each one of these boxes rather than hard coding them all in?

Reply:
Sure will do the refactor when I add the API integration. This is a v1.